File: 1362-closest-divisors/1362-closest-divisors.py
class Solution:
    def closestDivisors(self, num: int) -> List[int]:
        for i in range(int(math.sqrt(num + 2)), 1, -1):
            if (num + 1) % i == 0:
                return [i, (num + 1) // i]
            elif (num + 2) % i == 0:
                return [i, (num + 2) // i]
        
        return [1, num + 1]       
        

File: 1362-closest-divisors/1362-closest-divisors.swift
class Solution {
    func closestDivisors(_ num: Int) -> [Int] {
        var i = Int(sqrt(Double(num + 2)))
        
        while i >= 2 {
            if (num + 1) % i == 0 {
                return [i, (num + 1) / i]
            } else if (num + 2) % i == 0 {
                return [i, (num + 2) / i]
            }
            i -= 1
        }
        
        return [1, num + 1]
    }
}
